### ISSUE TITLE
Feedback on Protege 5.5

### ISSUE THREAD

**Issue by bblfish** [protegeproject/protege#963]
Dear Protegé,

I am currently working on exploring the Fido ontology with Protége 5.5 on OSX. I thought I'd post some feedback. I have 16 years experience with SemWeb technologies, though mostly around LinkedData, writing Java libraries to build apps (with nearly no inferencing), etc... So I thought my experience getting started with Protégé could be useful.

I find there is huge improvements in all of Protégé since I last tried using it 10 years ago. I love the history back and forth buttons. The tool works nicely, and does not freeze (or not for very long) as it used to.  So the following are just a few things that stalled me a bit. 

I am trying to play with [FIDO ontologies](https://github.com/edmcouncil/fibo/issues/1033) and Protege is very helpful here to make sense of it. There are a lot of classes there and so it is not possible really to keep that in ones head by reading the turtle directly, so a graphical display is very helpful. Also I am very interested in all the extra facts deduced by the DL reasoning engines.

So that brought me perhaps to the first problem, is choosing the reasoners. I think the only one that seems to work is the Hermit 1.4.3.456 reasoner. It is difficult to tell from the modal dialog
box that popped up after a while and shown below if that is the reasoner running (as I tried the Pellet incremental one first, which returned an error). It would be useful if the dialog box said what reasoner was running. Also I can't tell what will happens if I click the "cancel" button. Is reasoning going to continue? If it is going to continue that this should not be a modal dialog, since there is no reason for it to stop my exploration of the ontology and the button should say "continue working", not "cancel". Especially as it looks that for an ontology the size of FIDO the reasoning is going to take perhaps 2 hours or more. 

<img width="958" alt="Screen Shot 2020-06-13 at 12 54 52" src="https://user-images.githubusercontent.com/124506/84567396-b6a26c00-ad78-11ea-8da0-a04ac0491b37.png">

I think a few days ago I clicked "Cancel" and I guess it then continued reasoning. But I could
not be sure because in the "Active Ontolgoy tab" there is no information about how many triples are from the original data and how many are inferred.  Also it speaks of rules there and I am not sure if those correspond to triples. 

If it is possible to click "cancel" then one should be able to bring that box up again by clicking an inference button in the bottom right of Protege to see how inferencing is progressing. It should also tell one if inferencing is finished. 

Because I had trouble working out if the reasoner was working and how long it would take (perhaps because I clicked "Cancel") I started looking for other inferencing engines. I saw that [Stardog](https://stardog.com/) gives a two month free trial so I tried installing that. I found that loading FIBO into Stardog to be a bit problematic (see the [FIDO issue on loading Stardog](https://github.com/edmcouncil/fibo/issues/1036)). That is very likely a problem with Stardog. 
They have a nice UI for SPARQL querying their DB but their graph view is not that helpful for such a large ontology. It is clear that Protege is the tool needed to browse the ontology. It would be useful if one could add Stardog as a remote DB to work on from Protege. (and for them to make loading local and remote data by following links a drag and drop operation).

I think that is all for the moment. I'll add feedback if I come across other issues.

 
I have a 32GB of RAM laptop.  

For some reason now I got a Reasoner progress bar to appear, but it is what is called a [Modal Dialog Box](https://www.nngroup.com/articles/modal-nonmodal-dialog/) and stops all interaction.

**Comment by bblfish** [comment]
Another thing that would be useful would be to know what level of reasoning an ontology requires. I found a bug report that this was available at one point, but this is no longer the case. Such information should help estimate how much time a reasoner is going to take.

**Comment by bblfish** [comment]
It looks like it took something like 4 hours to classify. Then I was able to use Protege a bit. Now  the modal window has popped up again with the text "Realizing" (explained in the [HermiT reasoner document](https://hermit-reasoner.com/boris.motik/pubs/ghmsw14HermiT.pdf). Note the HermiT folks should get a free [LetsEncrypt](https://letsencrypt.org/) certificate so that people can get to the doc))
Again it is not clear if the Cancel button is going to cancel the task. I am going to have to risk this because I want to get on with my work. But if that is a mistake, then I don't know if I may have wasted 4 hours of previous compute time, or if there is a way to restart that task.
<img width="512" alt="Screen Shot 2020-06-13 at 16 54 55" src="https://user-images.githubusercontent.com/124506/84571888-9da8b380-ad96-11ea-8fd7-cd791df59590.png">

**Comment by bblfish** [comment]
So it is 20 minutes ago that the "Cancelled. Waiting for reasoner to terminate" popped up, stopping all my ability to work. Is this stuck? What happens if I force kill Protégé? Will the data then be all messed up? Will I then need to start from scratch? 
Does all of this need to block the whole of protégé? Can it not be run in a side thread with lower priority?

**Comment by bblfish** [comment]
Ah the "Cancel" Modal dialog just disappeared.

**Comment by bblfish** [comment]
So I saw a question mark button next to a subClassOf entity `lcc-lr:hasName` and intruiged I clicked it. This opened the modal dialog box below:

<img width="877" alt="Screen Shot 2020-06-13 at 17 20 09" src="https://user-images.githubusercontent.com/124506/84572454-5cb29e00-ad9a-11ea-8278-e88cd3cc7eb5.png">

Modal Dialog boxes should be used very sparingly as this article [Modal & Nonmodal Dialogs: When (& When Not) to Use Them](https://www.nngroup.com/articles/modal-nonmodal-dialog/)
 explains. I waited perhaps 5 minutes. Then I thought "well I am ok with just one explanation" as I just wanted to see what that button was for.  So I clicked "Stop Searching". Then it took another 5 minutes to close the dialog, a time during which I could not use Protégé. Then it opened an explanation box. I clicked on a button for "laconic explanation" wondering what that is and I got back to the "Computing explanations" modal dialog, which I am again waiting for to close. 

As a result of this, a normal user would start feeling very weary of ever clicking a "?" button. The more such problems appear the more unuseable Protégé gets. I really doubt it has to be this way.

**Comment by VladimirAlexiev** [comment]
FIDO -> FIBO.

@bblfish you are an extremely patient person ;-)

**Comment by tipluChris** [comment]
I agree with @bblfish's feedback. Protege is a great tool, but in terms of usability it has some things that could use improving. Would love to see these things improved! :)
